src/Covenant/Internal/Type.hs

```diff
@@ -151,6 +177,9 @@ instance Eq1 CompT where
   liftEq f (CompT abses1 xs) (CompT abses2 ys) =
     abses1 == abses2 && liftEq f xs ys
 
+newtype TyName = TyName Text
+  deriving (Show, Eq, Ord, IsString) via Text
```

Comment:
This will likely require `@since`s, as we are very likely to publically expose it.